recipes/nspr/all/conanfile.py

```diff
@@ -0,0 +1,165 @@
+        return self._autotools
+
+    def build(self):
+        with tools.chdir(self._source_subfolder):
+            with self._build_context():
+                autotools = self._configure_autotools()
+                autotools.make()
+
+    def package(self):
+        self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
+        with tools.chdir(self._source_subfolder):
+            with self._build_context():
+                autotools = self._configure_autotools()
+                autotools.install()
+        tools.rmdir(os.path.join(self.package_folder, "bin"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+        if self.settings.os == "Windows":
+            if self.options.shared:
+                os.mkdir(os.path.join(self.package_folder, "bin"))
+            for lib in self._library_names:
+                libsuffix = "lib" if self.settings.compiler == "Visual Studio" else "a"
+                libprefix = "" if self.settings.compiler == "Visual Studio" else "lib"
+                if self.options.shared:
+                    os.unlink(os.path.join(self.package_folder, "lib", "{}{}_s.{}".format(libprefix, lib, libsuffix)))
+                    os.rename(os.path.join(self.package_folder, "lib", "{}.dll".format(lib)),
+                              os.path.join(self.package_folder, "bin", "{}.dll".format(lib)))
+                else:
+                    os.unlink(os.path.join(self.package_folder, "lib", "{}{}.{}".format(libprefix, lib, libsuffix)))
+                    os.unlink(os.path.join(self.package_folder, "lib", "{}.dll".format(lib)))
+            if not self.options.shared:
+                tools.replace_in_file(os.path.join(self.package_folder, "include", "nspr", "prtypes.h"),
+                                      "#define NSPR_API(__type) PR_IMPORT(__type)",
+                                      "#define NSPR_API(__type) extern __type")
+                tools.replace_in_file(os.path.join(self.package_folder, "include", "nspr", "prtypes.h"),
+                                      "#define NSPR_DATA_API(__type) PR_IMPORT_DATA(__type)",
+                                      "#define NSPR_DATA_API(__type) extern __type")
+        elif self.settings.os == "Macos":
+            for lib in self._library_names:
+                if self.options.shared:
+                    os.unlink(os.path.join(self.package_folder, "lib", "lib{}.a".format(lib)))
+                else:
+                    os.unlink(os.path.join(self.package_folder, "lib", "lib{}.dylib".format(lib)))
+
+        if self.settings.compiler == "Visual Studio":
+            if self.settings.build_type == "Debug":
+                for lib in self._library_names:
+                    os.remove(os.path.join(self.package_folder, "lib", "{}.pdb".format(lib)))
+
+        if not self.options.shared or self.settings.os == "Windows":
```

Comment:
do we need chmod for Windows?

Reply:
Not needed per se, I just noticed these access permissions while using msys2 during development of this recipe.